### ISSUE TITLE
Always [0,1] mask on mutil-classes models

### ISSUE THREAD

**Issue by g-mero** [milesial/Pytorch-UNet#471]
expect: 
![124](https://github.com/milesial/Pytorch-UNet/assets/62990607/d619d969-f3ad-447c-b2b3-b4b09eaafb52)

but result:
![res](https://github.com/milesial/Pytorch-UNet/assets/62990607/997ec2b0-b50f-4741-aa7c-c3d85fe37d08)

train script
`python train.py -e 5 -b 10 -s 1 -v 10 -c 4 --amp`

predict script
`python predict.py -m checkpoints/checkpoint_epoch4.pth -s 1 -c 4 -i data/imgs/124.jpg -o res.png`

**Comment by g-mero** [comment]
cause by data under-fitting. improve dataset and increase epochs fix this

**Comment by BenjaminSchaar** [comment]
I have the same problem! I changed the color mapping to fit multiple classes but I only get 2 classes :(
